resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/v1/KubernetesRestProtocolMessages.scala

```diff
@@ -35,6 +37,9 @@ case class KubernetesCreateSubmissionRequest(
     driverPodKubernetesCredentials: KubernetesCredentials,
     uploadedJarsBase64Contents: TarGzippedData,
     uploadedFilesBase64Contents: TarGzippedData) extends SubmitRestProtocolRequest {
+  @JsonIgnore
+  override val messageType: String = s"kubernetes.v1.${Utils.getFormattedClassName(this)}"
```

Comment:
This is necessary due to the fact that `SubmitRestProtocolRequest` expects all subclasses to live in the package `org.apache.spark.deploy.rest`. The deserialization fails otherwise. This is a workaround to keep the package names consistent for now until we remove this class entirely.